File: sheet 8 N. Cakeminator/main.py
r ,c = list(map(int, input().split()))
arr_row = [0]*r
arr_col = [0]*c
res = 0
for i in range(r):
    cake = input()
    for j in range(c):
        if cake[j] == 'S':
            arr_row[i] = 1
            arr_col[j] = 1

for k in range(r):
    for y in range(c):
        if arr_row[k] == 0 or arr_col[y] == 0:
            res += 1

print(res)

File: sheet 8 N. Cakeminator/main.js
var input = readline().split` `.map(x => +x);
var rows = input[0], cols = input[1];
var cake = [];
var eatenCakes = 0;
for (var i = 0; i < rows; i++) {
    var row = readline().split``;
    cake.push(row);
    if (!row.includes("S")) {
        for (var j = 0; j < cols; j++) {
            cake[i][j] = '#';
            eatenCakes++;
        }
    }
}


for (var i = 0; i < cols; i++) {
    var col = [];
    for (var j = 0; j < rows; j++) {
        col.push(cake[j][i]);
    }
    if (!col.includes('S')) {
        for (var j = 0; j < rows; j++) {
            if (cake[j][i] === '.') {
                cake[j][i] = '#';
                eatenCakes++;
            }
        }
    }
}
print(eatenCakes);
